ts/pages/staking/home.tsx

```diff
@@ -10,44 +11,23 @@ import { StakingConfirmationDialog } from 'ts/components/dialogs/staking_confirm
 import { StakingPageLayout } from 'ts/components/staking/layout/staking_page_layout';
 import { StakingPoolDetailRow } from 'ts/components/staking/staking_pool_detail_row';
 
-import { ScreenWidths } from 'ts/types';
-
 import { StakingHero } from 'ts/components/staking/hero';
 import { Heading } from 'ts/components/text';
-
-const stakingPools = [
-    {
-        id: '29n5c290cn0cc2943cn239',
-        name: 'Staking 01',
-        thumbnailUrl: 'https://cdn.worldvectorlogo.com/logos/0x-virtual-money-.svg',
-        feesCollectedEth: 0.03281,
-        stakingPercent: 2000,
-        rewardsSharePercent: 100,
-        location: '234 CALIFORNIA',
-    },
-    {
-        id: '29n5c290cn0cc2943cn230',
-        name: 'Staking 02',
-        thumbnailUrl: 'https://cdn.worldvectorlogo.com/logos/0x-virtual-money-.svg',
-        feesCollectedEth: 0.03281,
-        stakingPercent: 2000,
-        rewardsSharePercent: 100,
-        location: '234 CALIFORNIA',
-    },
-    {
-        id: '29n5c290cn0cc2943cn240',
-        name: 'Staking 03',
-        thumbnailUrl: 'https://cdn.worldvectorlogo.com/logos/0x-virtual-money-.svg',
-        feesCollectedEth: 0.03281,
-        stakingPercent: 2000,
-        rewardsSharePercent: 100,
-        location: '234 CALIFORNIA',
-    },
-];
+import { PoolWithStats, ScreenWidths } from 'ts/types';
+import { apiClient } from 'ts/utils/api_client';
 
 export interface StakingIndexProps {}
 export const StakingIndex: React.FC<StakingIndexProps> = props => {
     const [isStakingConfirmationOpen, toggleStakingConfirmation] = React.useState(false);
+    const [stakingPools, setStakingPools] = React.useState<PoolWithStats[] | undefined>(undefined);
+    React.useEffect(() => {
+        const fetchAndSetPoolsAsync = async () => {
+            const resp = await apiClient.getStakingPoolsAsync();
+            setStakingPools(resp.stakingPools);
+        };
+        // tslint:disable-next-line:no-floating-promises
+        fetchAndSetPoolsAsync();
+    });
```

Comment:
Correct me if I'm wrong but `useEffect` with no second arg would trigger on every component render and re-fetch the data. We would rather fetch it only once on page load right?
```suggestion
    }, []);
```

Reply:
Oops!